react-components/src/components/RevealContainer/RevealResourcesContext.tsx

```diff
@@ -0,0 +1,38 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import { type CogniteModel } from '@cognite/reveal';
+import { type ReactElement, createContext, useContext, useState, type ReactNode } from 'react';
+
+type RevealResourcesContextType = {
+  reveal3DResources: CogniteModel[];
+  update3DResources: (newResources: CogniteModel[]) => void;
+};
+
+export const RevealResourcesContext = createContext<RevealResourcesContextType | null>(null);
+
+export const useRevealResources = (): RevealResourcesContextType => {
+  const element = useContext(RevealResourcesContext);
+  if (element === null) {
+    throw new Error('useRevealResources must be used within a RevealResourcesContextProvider');
+  }
+  return element;
+};
+
+export const RevealResourcesContextProvider = ({
+  children
+}: {
+  children: ReactNode;
+}): ReactElement => {
+  const [reveal3DResources, setRevealResources] = useState<CogniteModel[]>([]);
+
+  const update3DResources = (newResources: CogniteModel[]): void => {
+    setRevealResources(newResources);
+  };
```

Comment:
I don't think we need a wrapper for `setRevealResources` it can be used directly in the context.

Reply:
Also, name `update3DResources` is a little more ambiguous in my opinion, so having it as `setRevealResources` is clear in terms of the fact that it sets the whole thing, not adding resources to already present state.